File: solutions/go/largest-series-product/1/largest_series_product.go
package lsproduct

import (
	"errors"
)

func LargestSeriesProduct(digits string, span int) (int64, error) {
	var result int64
	result = 0
	if span > len(digits) {
		return 0, errors.New("span must be smaller than string length")
	}
	if span < 0 {
		return 0, errors.New("span must be greater than zero")
	}

	for i := 0; i < len(digits)-span+1; i++ {
		var product int64
		product = 1
		for j := 0; j < span; j++ {
			if (digits[i+j] - '0') > 9 {
				return 0, errors.New("digits must only contain digits")

			}
			product *= int64(digits[i+j] - '0')
		}
		if product > result {
			result = product
		}
	}
	return result, nil
}

File: solutions/typescript/largest-series-product/1/largest-series-product.ts
export const largestProduct = (inputString: string, span: number) => {

  if (span < 0) {
    throw new Error('Span must not be negative')
  }

  if (span > inputString.length) {
    throw new Error('Span must be smaller than string length')
  }

  if (inputString.match(/\D/)) {
    throw new Error('Digits input must only contain digits')
  }

  if (span == inputString.length) {
    return inputString.split('').reduce((acc, digit) => acc * parseInt(digit), 1)
  }
  let maxProduct = 0
  for (let i = 0; i <= inputString.length - span; i++) {
    let product = inputString.slice(i, i + span).split('').reduce((acc, digit) => acc * parseInt(digit), 1)
    if (product > maxProduct) {
      maxProduct = product
    }
  }

  return maxProduct
}
